docs/Codebase.md

```diff
@@ -0,0 +1,93 @@
+## <a href="#codebase">Codebase</a>
+InsertionsClient solution consists of 2 projects for the application logic and 2 projects for their respective tests.
+
+Besides the solution, the documentation and the `.yml` file for the build pipeline also reside in the repo at https://github.com/dotnet/insertions-client.
+
+This section covers the highlights of both `Core` and the `Console` projects as well as the `.yml` file for the build pipeline.
+
+**Table of Contents**
+- [Core Project](#core-project)
+  - [IInsertionApi Interface](#iinsertionapi-interface)
+  - [InsertionApi Class](#insertionapi-class)
+- [Console Project](#console-project)
+  - [Program Class](#program-class)
+  - [Logging](#logging)
+- [Build Pipeline](#build-pipeline)
+
+
+### [Core Project](#core-project)
+Core is the project where the main insertion logic resides in. The project exposes an interface (`IInsertionApi`) so that the code can be consumed by other projects and invoked like an API. 
```

Comment:
consider:
Core exposes interface `IInsertionApi` which defines insertion features.

Reply:
I can't tell which part of the proposed change are you improving here. Can you elaborate?

For instance:
My change is better because, the word "exposes" is used in a wrong position, the word "invoked" doesn't sound applicable to this case or the sentence is too long etc.